src/main/java/io/vertx/mqtt/MqttClientOptions.java

```diff
@@ -202,7 +204,16 @@ public String getWillTopic() {
   /**
    * @return will message content
    */
+  @Deprecated
   public String getWillMessage() {
+    return willMessage.toString(StandardCharsets.UTF_8);
+  }
+
+  /**
+   * @return will message bytes content
+   */
+  @GenIgnore
```

Comment:
Please mark the `willMessage` property with `@GenIgnore` instead of this one.

Then in the constructor which takes a `JsonObject`, add some code to take into account `willMessage` field, if present (transform to Buffer and log warning)

Reply:
Ok, i made these changed. However, https://github.com/vert-x3/vertx-mqtt/pull/244#discussion_r1372743872 is now reintroduced.